app/components/publications/overview/publications-table-row.js

```diff
@@ -53,13 +61,44 @@ export default class PublicationsTableRowComponent extends Component {
     return this.publicationStatus && getPublicationStatusPillStep(this.publicationStatus);
   }
 
+  /** @returns {Date?} undefined if no translation-activities */
+  async getTranslationRequestDate(publicationFlow) {
+    const publicationSubcase = await publicationFlow.translationSubcase;
+    const translationActivities = await publicationSubcase.translationActivities;
+    const requestDates = translationActivities.mapBy('startDate');
+    const mostRecentDate = this.getMaxDate(requestDates);
+    return mostRecentDate;
+  }
+
+  /** @returns {Date?} undefined if no ProofingActivities */
   async getProofRequestDate(publicationFlow) {
     const publicationSubcase = await publicationFlow.publicationSubcase;
     const proofingActivities = await publicationSubcase.proofingActivities;
-    const startDates = proofingActivities.mapBy('startDate');
-    startDates.sort();
-    const firstProofRequestDate = startDates[0];
-    return firstProofRequestDate;
+    const requestDates = proofingActivities.mapBy('startDate');
+    const mostRecentDate = this.getMaxDate(requestDates);
+    return mostRecentDate;
+  }
+
+  getMaxDate(dates) {
+    // default .sort() of Date objects does not give expected results
+    dates = dates.sortBy();
+    const max = dates.lastObject;
+    return max;
+  }
+
+  /** @returns {Date?} undefined if no ProofingActivities or a ProofingActivity is not finished */
+  async getProofReceivedDate(publicationFlow) {
+    const publicationSubcase = await publicationFlow.publicationSubcase;
+    const proofingActivities = await publicationSubcase.proofingActivities;
+    let receivedDates = proofingActivities.mapBy('endDate');
+    // default .sort() of Date objects does not give expected results
+    // JS sorts `undefined` after Date objects
+    // if there is a ProofingActivity which is not finished (endDate === undefined)
+    // undefined will be the .lastObject
+    // not using Embers .sortBy() to avoid sort algorithm changes
```

Comment:
Wat bedoel je hiermee?

Reply:
Omdat sortBy() anders werkt voor enkel `dates`, dacht ik dat sorteren op `dates` en `undefineds` misschien kan wijzigen in een volgende Ember versie.